tests/unit/runners/test_filequeue.py

```diff
@@ -0,0 +1,172 @@
+
+import tempfile
+import shutil
+import os
+import pytest
+import time
+from mock import patch, MagicMock
+from srv.modules.runners import filequeue
+
+class TestFileQueue():
+    '''
+    This class tests the FileQueue class for the salt runner
+    '''
+    def test_default_queue(self):
+        '''
+        Verify that the default queue exists
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath)
+        queue = os.path.isdir("{}/default".format(dirpath))
+        shutil.rmtree(dirpath)
+        assert queue == True
+
+    def test_named_queue(self):
+        '''
+        Verify that the 'lunch' queue exists
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath, queue='lunch')
+        queue = os.path.isdir("{}/lunch".format(dirpath))
+        shutil.rmtree(dirpath)
+        assert queue == True
+
+    def test_dirs(self):
+        '''
+        Verify that dirs returns the queues
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath, queue='lunch')
+        names = fq.dirs()
+        shutil.rmtree(dirpath)
+        assert 'lunch' == names[0]
+
+    @patch('srv.modules.runners.filequeue.FileQueue._fire_event', autospec=True)
+    def test_touch(self, fire_event):
+        '''
+        Verify that touch creates a file
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath)
+        touched = fq.touch("red")
+        exists = os.path.isfile("{}/default/red".format(dirpath))
+        shutil.rmtree(dirpath)
+        assert touched and exists 
+
+    @patch('srv.modules.runners.filequeue.FileQueue._fire_event', autospec=True)
+    def test_ls(self, fire_event):
+        '''
+        Verify that ls returns filenames
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath)
+        touched = fq.touch("red")
+        touched = fq.touch("blue")
+        files = fq.ls()
+        shutil.rmtree(dirpath)
+        assert files == ['red', 'blue']
```

Comment:
Getting a test failure here.
```
assert ['blue', 'red'] == ['red', 'blue']
```

So its just an ordering issue. Either sort both lists before the comparison or compare sets.